bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/dto/ActionTypeDTOMapper.java

```diff
@@ -29,15 +31,28 @@ public static ActionTypeDTO map(final ActionType actionType) {
         return actionTypeDto;
     }
 
+    public static CompositeActionTypeDTO map(final CompositeActionType actionType) {
+        final CompositeActionTypeDTO actionTypeDto = new CompositeActionTypeDTO();
+        fillProperties(actionType, actionTypeDto);
+        actionTypeDto.inputs = actionType.getInputs();
+        actionTypeDto.outputs = actionType.getOutputs();
+        actionTypeDto.children = ActionDTOMapper.map(actionType.getChildren());
+        return actionTypeDto;
+    }
+
     public static List<ActionTypeDTO> map(final Collection<ActionType> types) {
         if (types == null) {
             return null;
         }
         final List<ActionTypeDTO> dtos = new ArrayList<ActionTypeDTO>(types.size());
         for (final ActionType type : types) {
-            dtos.add(map(type));
+            if (type instanceof CompositeActionType) {
+                dtos.add(map((CompositeActionType) type));
+            } else {
+                dtos.add(map(type));
+            }
         }
         return dtos;
     }
 
-}
+}
```

Comment:
Please add a new line after `}`. This will IMHO work better with Git (according to POSIX ever `line` ends with a new line character).